ui/shared/components/panel/view-fields/BaseFieldView.jsx

```diff
@@ -18,76 +20,126 @@ const FieldValue = styled.div`
+          buttonText={this.props.editLabel}
+          editIconName={this.props.editIconName}
+          onSubmit={this.props.onSubmit}
+          initialValues={this.props.initialValues}
+          formFields={this.props.formFields}
+          formContainer={<div style={this.props.modalStyle} />}
+          showErrorPanel={this.props.showErrorPanel}
+          confirmDialog={this.props.addConfirm}
+          size="tiny"
+        />
+      )
+
+    const editButton = this.props.isEditable && (this.props.formFields ?
+      updateButton
+      : (
+        <DispatchRequestButton
+          key="edit"
+          buttonContent={<Icon link size="small" name="plus" />}
+          onSubmit={() => this.props.onSubmit(this.props.initialValues)}
+          confirmDialog={this.props.addConfirm}
+        />
+      ))
+
+    const deleteButton = this.props.isDeletable && (
+      <DeleteButton
+        key="delete"
+        initialValues={this.props.initialValues}
+        onSubmit={this.props.onSubmit}
+        confirmDialog={this.props.deleteConfirm}
+        size="tiny"
       />
-    ))
-  const deleteButton = props.isDeletable && (
-    <DeleteButton
-      key="delete"
-      initialValues={props.initialValues}
-      onSubmit={props.onSubmit}
-      confirmDialog={props.deleteConfirm}
-      size="tiny"
-    />
-  )
-  const buttons = [editButton, deleteButton]
+    )
+    const buttons = [editButton, deleteButton]
+
+    return (
+      <span style={this.props.style || {}}>
+        {this.props.isPrivate && <StaffOnlyIcon />}
+        {this.props.fieldName && [
+          <b key="name">{this.props.fieldName}{hasValue ? ':' : null}<HorizontalSpacer width={10} /></b>,
+          ...buttons,
+          this.props.compact && (buttons.some(b => b) ? <HorizontalSpacer width={10} key="hs" /> : null),
+          !this.props.compact && <br key="br" />,
+        ]}
+        {
+          hasValue && !this.props.hideValue && this.state.showInLineButton &&
+          <FieldValue compact={this.props.compact} fieldName={this.props.fieldName}>
+            {this.props.fieldDisplay(fieldValue, this.props.compact, fieldId)}
+          </FieldValue>
+        }
+        {!this.props.fieldName && buttons}
+      </span>)
+  }
 
-  return (
-    <span style={props.style || {}}>
-      {props.isPrivate && <StaffOnlyIcon />}
-      {props.fieldName && [
-        <b key="name">{props.fieldName}{hasValue ? ':' : null}<HorizontalSpacer width={10} /></b>,
-        ...buttons,
-        props.compact && (buttons.some(b => b) ? <HorizontalSpacer width={10} key="hs" /> : null),
-        !props.compact && <br key="br" />,
-      ]}
-      {
-        hasValue && !props.hideValue &&
-        <FieldValue compact={props.compact} fieldName={props.fieldName}>
-          {props.fieldDisplay(fieldValue, props.compact, fieldId)}
-        </FieldValue>
-      }
-      {!props.fieldName && buttons}
-    </span>)
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
you should use `React.PureComponent` instead of `React.Component` and then you won't need this function at all